full-stack-tests/presentation/src/learning-snippets/Hierarchies.test.ts

```diff
@@ -24,305 +28,1613 @@ describe("Learning Snippets", () => {
-      expect(nodes).to.containSubset([{
-        label: { displayValue: "B" },
-      }, {
-        label: { displayValue: "A" },
-      }]);
+        };
+        // __PUBLISH_EXTRACT_END__
+        printRuleset(ruleset);
+
+        // The root node rule meets schema requirement, but only the first sub-condition's condition
+        // attribute evaluates to `true` - expect only the "A" node.
+        const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+        expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+          label: { displayValue: "A" },
+        }]);
+      });
+
+      it("uses `autoExpand` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ Hierarchies.AutoExpand.Ruleset
+        // The ruleset defines a root node "A" which should be automatically expanded. The flag is only
+        // set if the node actually has children.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.RootNodes,
+            autoExpand: true,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.CustomNode,
+              type: "A",
+              label: "A",
+            }],
+          }, {
+            ruleType: RuleTypes.ChildNodes,
+            condition: `ParentNode.Type = "A"`,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.CustomNode,
+              type: "B",
+              label: "B",
+            }],
+          }],
+        };
+        // __PUBLISH_EXTRACT_END__
+        printRuleset(ruleset);
+
+        // The root node is expected to have `isExpanded = true`
+        const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+        expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+          label: { displayValue: "A" },
+          isExpanded: true,
+        }]);
+      });
+
     });
 
-    it("uses `onlyIfNotHandled` attribute", async () => {
-      // __PUBLISH_EXTRACT_START__ Hierarchies.OnlyIfNotHandled.Ruleset
-      // The ruleset has two root node rules that return nodes "A" and "B" respectively. The "A" rule has
-      // lower priority and `onlyIfNotHandled` attribute, which allows it to be overriden by higher priority rules.
-      // The "B" rule does exactly that.
-      const ruleset: Ruleset = {
-        id: "example",
-        rules: [{
-          ruleType: RuleTypes.RootNodes,
-          priority: 1,
-          onlyIfNotHandled: true,
-          specifications: [{
-            specType: ChildNodeSpecificationTypes.CustomNode,
-            type: "A",
-            label: "A",
+    describe("Specifications", () => {
+
+      it("uses `hideNodesInHierarchy` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ Hierarchies.Specification.HideNodesInHierarchy.Ruleset
+        // The ruleset contains a root node specification for `bis.PhysicalModel` nodes which are grouped by class and hidden. This
```

Comment:
Suggestion: Since were giving examples about specifications here, which are always in contained in rules, maybe we can skip the repeating "The ruleset contains..."

Maybe it would be better with something along the lines of:
Specification returns `bis.PhysicalModel` nodes which are grouped by class and hidden.
...

Reply:
I understand what you're saying, but the suggested change isn't correct - the specifications are just specifications and neither they, not rulesets or rules, return anything. The presentation rules library returns nodes after evaluating the ruleset. So while I'd like to shorten the description and avoid repeating things, I don't think we can just say that they return something just to make that happen.